Source/ForecastDeskbarView.cpp

```diff
@@ -108,14 +112,11 @@ ForecastDeskbarView::MessageReceived(BMessage* message)
 {
 	if (message->what == kUpdateForecastMessage) {
 		BString weatherDetailsText;
-		weatherDetailsText << "Temperature: "
-						   << FormatString(fForecastView->Unit(),
-								  fForecastView->Temperature())
-						   << "\n";
-		weatherDetailsText << "Condition: " << fForecastView->GetCondition()
-						   << "\n";
-		weatherDetailsText << "Location: " << fForecastView->CityName();
-		SetToolTip(weatherDetailsText.String());
+		weatherDetailsText.SetToFormat(B_TRANSLATE("Temperature: %s\nCondition: %s\nLocation: %s"),
+			FormatString(fForecastView->Unit(), fForecastView->Temperature()).String(),
+			fForecastView->GetStatus().String(),
```

Comment:
Here the "condition" (sunny, rainy, ...) was replaced by the status (loading, up to date). I guess this should be reverted, but I don't know if there is a way to convert the condition into a string.

Reply:
Sorry for  the late reply :(
`fForecastView->GetStatus()` will return the status (loading, no network, connection error), but also the correct condition (sunny, rainy etc.) as a translatable string when available.
I'm puzzled as to why @humdingerb isn't getting the correct forecast. On my machine it works (both VM and natively), but after changing the code I have to open the app, enable and disable the Desktop replicant, and then restart the app to make sure the replicant is running the new code. I also uninstalled the Weather app from HaikuDepot to avoid any conflicts.

![screenshot2](https://github.com/HaikuArchives/Weather/assets/12958546/9dcbafd2-3642-4d11-8f87-c5dd9d9c5551)

Reply:
No idea either. I checked out PR #103, built cleanly and there is no other Weather binary on the system. Still the same: the Deskbar replicant always shows the sun and "Condition" says "Loading...".